src/test/java/com/google/sps/UserServletTest.java

```diff
@@ -0,0 +1,311 @@
+    toggleLogin("another@example.com");
+    when(request.getParameter("event-name")).thenReturn("Book Drive");
+    when(request.getParameter("event-description")).thenReturn("Let's donate books for kids");
+    when(request.getParameter("street-address")).thenReturn("School Drive");
+    when(request.getParameter("city")).thenReturn("Los Angeles");
+    when(request.getParameter("state")).thenReturn("California");
+    when(request.getParameter("date")).thenReturn("2020-17-05");
+    when(request.getParameter("start-time")).thenReturn("10:00");
+    when(request.getParameter("all-tags")).thenReturn("['education']");
+    testEventServlet.doPost(request, response);
+
+    // logout
+    toggleLogin("another@example.com");
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    helper.setUp();
+    PowerMockito.mockStatic(UserServiceFactory.class);
+    mockService = new MockedUserService();
+    when(UserServiceFactory.getUserService()).thenReturn(mockService);
+    testEventServlet = new EventServlet();
+    testAuthServlet = new AuthServlet();
+    testUserServlet = new UserServlet();
+
+    // get the initial login url
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+    StringWriter out = new StringWriter();
+    PrintWriter writer = new PrintWriter(out);
+    when(response.getWriter()).thenReturn(writer);
+    testAuthServlet.doGet(request, response);
+    out.flush();
+
+    LoginObject result = gson.fromJson(out.toString(), LoginObject.class);
+    activeUrl = result.url;
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    activeUrl = "";
+  }
+
+  @Test
+  public void notLoggedIn() throws IOException {
+    // post the events to datastore
+    postEventsSetup();
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+    StringWriter out = new StringWriter();
+    PrintWriter writer = new PrintWriter(out);
+    when(response.getWriter()).thenReturn(writer);
+
+    testUserServlet.doGet(request, response);
+    out.flush();
+    List<Entity> resultingEntities = gson.fromJson(out.toString(), new TypeToken<ArrayList<Entity>>(){}.getType());
+
+    // create the expected resulting search results
+    Entity goalEntity = new Entity("Event");
+    goalEntity.setProperty("eventName", "Lake Clean Up");
+    goalEntity.setProperty("eventDescription", "We're cleaning up the lake");
+    goalEntity.setProperty("streetAddress", "678 Lakeview Way");
+    goalEntity.setProperty("city", "Lakeside");
+    goalEntity.setProperty("state", "Michigan");
+    goalEntity.setProperty("date", "2020-17-05");
+    goalEntity.setProperty("startTime", "14:00");
+    goalEntity.setProperty("endTime", "");
+    goalEntity.setProperty("coverPhoto", "");
+    goalEntity.setProperty("tags", "['environment']");
+    goalEntity.setProperty("creator", "test@example.com");
```

Comment:
Would be good to refactor into a helper method. This way, the method can be reused in the next test below.

Reply:
Done!